src/components/CounterSegment.js

```diff
@@ -27,100 +50,66 @@ const PERIOD_LIMITS = {
  *   digitWrapper={(digit) => digit}
  * />
  */
-class CounterSegment extends React.Component {
+export default function CounterSegment (props) {
```

Comment:
It's just basic React stuff. If you don't need things like constructor or lifecycle methods, your component can be a function that returns whatever `render` of a class component would return. This is significantly more performant.